### ISSUE TITLE
Crash Visual Stuio after select json in Model Builder

### ISSUE THREAD

**Issue by dmamel** [dotnet/machinelearning-modelbuilder#1701]
Hi. I ned help.
Crash and close Visual Stuio after select json in Model Builder (object detection).
[json.zip](https://github.com/dotnet/machinelearning-modelbuilder/files/7006912/result.zip)

ML.Net Model Builder v16.6.3.2136603
Visual Studio 19 v16.11.0

![Snap 2021-08-18 at 14 46 17](https://user-images.githubusercontent.com/45320291/129892704-0354a78a-b87f-42bd-923e-423c091b3a95.png)

![Snap 2021-08-18 at 14 32 18](https://user-images.githubusercontent.com/45320291/129891773-49cd1695-e98a-453c-8236-3fecb172f251.png)

**Comment by LittleLittleCloud** [comment]
Can we take a look at your json file? Is that vott format?

**Comment by LittleLittleCloud** [comment]
Label it as Priority 0 because it crashes VS.

**Comment by dmamel** [comment]
yes, I have attached a file

![image](https://user-images.githubusercontent.com/45320291/130288829-410fdd42-b7c7-4dd2-b598-f7ffc3e6da27.png)

**Comment by LittleLittleCloud** [comment]
Thanks @dmamel 

It seems that model builder will crash in object detection when dataset only contains one image. We will fix it ASAP and in the meanwhile, one workaround can be simply adding more images.

**Comment by dmamel** [comment]
This is just an example. The minimal file for debugging. I tried different files, and 100 and 1000 records.
Here is an example of 11 entries.
[result (2).zip](https://github.com/dotnet/machinelearning-modelbuilder/files/7079509/result.2.zip)
![image](https://user-images.githubusercontent.com/45320291/131402976-7a53cf22-146f-4698-82d9-cc5637964ee9.png)

**Comment by LittleLittleCloud** [comment]
@dmamel 
I C, can I have the images as well so I can precisely reproduce the error

**Comment by dmamel** [comment]
Yes, sure
[Downloads.zip](https://github.com/dotnet/machinelearning-modelbuilder/files/7079614/Downloads.zip)

**Comment by LittleLittleCloud** [comment]
@dmamel 

It works fine on my PC
![image](https://user-images.githubusercontent.com/16876986/131547660-7629d9ef-3a6b-4dd5-a876-40daeb21fe54.png)
![image](https://user-images.githubusercontent.com/16876986/131547710-1fef58b3-f74c-4c34-adb6-5019739d666d.png)

It could be related with system setting (different decimal separator in English/Russian). I will switch to the Russian language to see if I can reproduce it, in the meanwhile, could you also try the English language to see if it works?

**Comment by dmamel** [comment]
Yes, everything works with the English regional settings (though not the first time, but it works).
Thanks))

**Comment by LittleLittleCloud** [comment]
Thanks @dmamel for the quick response, we will start fixing this shortly

**Comment by dmamel** [comment]
Everything works, but this result takes a long time to process. Is this how it should be?
In the Model Builder and in the generated code.
I thought that using a ready-made model should be fast.

![image](https://user-images.githubusercontent.com/45320291/131631003-346db045-5dea-4297-97a4-b1af7b9dabdd.png)

**Comment by LittleLittleCloud** [comment]
@dmamel 
Is that the first image to predict? If so it's normal as it takes some time to load the model

**Comment by LittleLittleCloud** [comment]
The model won't be loaded until it gets called (as per se lazy loading). So the first predict will be slow, but you should see the second and following prediction much faster.

**Comment by LittleLittleCloud** [comment]
@dmamel 
Hmmm thanks, we will look into it

**Comment by LittleLittleCloud** [comment]
#1760

**Comment by LittleLittleCloud** [comment]
@dmamel 

What CPU do you have

**Comment by dmamel** [comment]
Intel Core i5 1.7GHz 4CPUs 8Gb

![image](https://user-images.githubusercontent.com/45320291/131897989-f8c5ed77-26c7-412f-aa49-aae1ab70ea02.png)


Intel Core i7 3GHz 8CPUs 32Gb

![image](https://user-images.githubusercontent.com/45320291/131897911-52f0f846-f0f2-456f-89c5-28640dcf6b84.png)

**Comment by LittleLittleCloud** [comment]
@dmamel 

We confirmed from our vision team that seconds can be considered normal for a single image inference. And we are looking into batching inference solution to make it faster on GPU

**Comment by dmamel** [comment]
OK thanks

**Comment by Willyb30** [comment]
Hello
I have the same problem : as soon as my first 4 images are loaded and previewes in model builder, VS19 crashes.
I'm running a french OS. I saw changing regional settings to english could solve the problem ; 
Did you change VS settings or windows' ?

VS19 Version 16.11.2
i7 8700H

**Comment by dmamel** [comment]
Windows

**Comment by Willyb30** [comment]
I've changes Windows regional settings.
VS 19 now crashes after clicking "next step" in Azure configuration screen after a small loading time, right before showing Data panel

**Comment by LittleLittleCloud** [comment]
@Willyb30 

What system language do you have. You probably would still meet this question if your system language is still French.

**Comment by Willyb30** [comment]
Yes it is

**Comment by Willyb30** [comment]
I've installed english language settings and it now seems to work

Thank you

**Comment by naster01** [comment]
My system was in english but my region in french. Switched to english and now I can import my json data.

**Comment by vzhuqin** [comment]
Not repro this issue on latest main build: 16.7.6.2147403
- Microsoft Visual Studio Enterprise 2019: 16.11.3
- ML.Net Model Builder: 16.7.6.2147403
- Windows OS: Windows-10-Enterprise-21H1
- English System with Russia Region
![image.png](https://images.zenhubusercontent.com/60653e492390a08ea616cb10/5ef41cd1-e075-47e6-9c65-f65f87c40600)
- VS Language: Russian
![image.png](https://images.zenhubusercontent.com/60653e492390a08ea616cb10/47c33cc5-ba1f-443b-85e6-f61544f34193)
Can input json file and complete training.
![image.png](https://images.zenhubusercontent.com/60653e492390a08ea616cb10/f6568a5b-3796-4623-b335-2ae8be8f6242)
![image.png](https://images.zenhubusercontent.com/60653e492390a08ea616cb10/945b2de4-42a2-4b92-85e3-593ca58df47e)
